tests/to_unit_test.py

```diff
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+import pytest
+import scipp as sc
+
+
+def test_to_unit():
+    var = sc.scalar(1, unit='m')
+    assert sc.to_unit(var, unit='mm').unit == sc.Unit('mm')
+    assert sc.to_unit(var, unit=sc.Unit('mm')).unit == sc.Unit('mm')
+    with pytest.raises(sc.UnitError):
+        sc.to_unit(var, unit='abcdef')  # does not parse
+    with pytest.raises(TypeError):
+        sc.to_unit(var, unit=5)  # neither str nor Unit
+
+
+@pytest.mark.parametrize('year', [1800, 1911, 1956, 1984, 2001, 2022, 2036, 2100, 2467])
+def test_datetime64_with_explicit_ns_unit_is_consistent_irrespective_of_decimals(year):
+    assert sc.identical(sc.datetime(f'{year}-01-01T12:00:00', unit='ns'),
+                        sc.datetime(f'{year}-01-01T12:00:00.000', unit='ns'))
+    assert sc.identical(sc.datetime(f'{year}-01-01T12:00:00', unit='ns'),
+                        sc.datetime(f'{year}-01-01T12:00:00.000000', unit='ns'))
+    assert sc.identical(sc.datetime(f'{year}-01-01T12:00:00', unit='ns'),
+                        sc.datetime(f'{year}-01-01T12:00:00.000000000', unit='ns'))
```

Comment:
Not sure I understand the purpose of this test. Why does it compare to `'{year}-01-01T12:00:00'` in every case instead of  `'{year}-01-01T12:00:00.000000000'`?

Reply:
You mean you would compare, e.g., `{year}-01-01T12:00:00.000000000` to `{year}-01-01T12:00:00.000`? I think we could do that *in addition*, but why instead?

Reply:
Ultimately, it does not matter because the 3 assertions ensure that all cases are consistent. But I was surprised that the conversion from s to ns seems to be the reference that the others are compared to.

Reply:
You are right, it makes more sense if I convert to the precision of the (more precise) string in all cases. Will push an update.

Reply:
The test name doesn't make sense anymore. Also, I thought the point was to check that we can convert to ns correctly, regardless of the input unit.

Reply:
Changed again.

Reply:
Ok